docs/index.md

```diff
@@ -1 +1,100 @@
-# TODO
+## About
+
+**Blender Image Preview**, or BIP for short is a **[Blender](https://blender.org)**
+library by 3D Ninjas that allows addon developers to have extra functionality
+that comes built into Blender but takes quite some time to set up.
+
+So what exactly is the `.bip` format? The accronym is exactly the same as the
+name of the library **Blender Image Preview**. Its a format designed to be 100%
+compatible with the Blender in-memory format of these objects. It primarily
+guarantees, that we can just read the data from the file and copy it without any
+modifications in the right buffer.
+
+### Origin
+
+BIP came about as a solution to image previews on an asset manager that we are
+currently developing at **[3D Ninjas](https://3dninjas.io)**. While working on
+this project, we realized that the resolution of the images provided by the
+default `bpy.utils.previews` was quite limiting in that we had no control as to
+image quality/resolution. After some sessions and development, BIP came to life.
+We decided that such knowledge should not be hoarded but rather shared with the
+Blender community.
+
+### Ease-Of-Use
+
+The library can be used as a direct drop-in replacement for the Blender's
+default `bpy.utils.previews` module, however, it pays off to take a bit of time
+to look into the other features that come with the library as they are what make
+it stand out.
+
+---
+
+## Use Cases
+
+### Controlled Environment
+
+Quickly load optimized images images that they have control over into Blender
+using a custom format named `.bip`. How to convert your images to `.bip` is
+covered in the **[Getting Started](getting_started.md)** section.
+
+### Arbitrarily Sized Images
+
+Loading arbitrarily sized images. By default Blender's standard previews
+come at a resolution of `32px by 32px` for icons and `256px by 256px` for
+images. With BIP, you have control of the image sizes per collection. For
+example, you can choose to have a hero image previewed at a specific place
+within your addon of `1024px by 1024px`, you can then choose to have a more
+optimized collection of images `128px by 128px` to use for displaying items in
+a library.
+
+### Uncontrolled Environment
+
+Use of Python's [`Pillow`](https://pypi.org/project/Pillow/) library to
+quickly process images that you do not have control over. A use case for this
+would be if your addon allows for users to load their own images, you would
+not have control as to how big they. In such a case, you should prompt the user
+to download [`Pillow`](https://pypi.org/project/Pillow/), if this is not done,
+then the library will default to Blender's standard preview service.
```

Comment:
Point out, that the library has a built-in operator which can be used to install Pillow.

Reply:
Updated.